src/pipeline.ts

```diff
@@ -489,14 +551,483 @@ export async function scanRssFeeds() {
+		INCOMPLETE_SEARCHEES: 0,
+		FAILED: 0,
+		UNMATCHED: 0,
+		FOUND_BAD_FORMAT: false,
+		FLAT_LINKING: flatLinking,
+	};
+	let count = 0;
+	for (const torrentFilePath of dirContents) {
+		const progress = chalk.blue(`(${++count}/${dirContents.length})`);
+		const filePathLog = getTorrentFilePathLog(torrentFilePath);
+		if (shouldCleanUpTorrent(torrentFilePath)) {
+			toDelete.add(torrentFilePath);
+		}
+		let meta: Metafile;
+		try {
+			meta = await parseTorrentFromFilename(torrentFilePath);
+		} catch (e) {
+			logger.error({
+				label: Label.INJECT,
+				message: `${progress} Failed to parse ${filePathLog}`,
+			});
+			logger.debug(e);
+			continue;
+		}
+		const metaLog = getLogString(meta, chalk.bold.white);
+
+		const torrentNameInfo =
+			await parseInfoFromSavedTorrent(torrentFilePath);
+		const tracker = torrentNameInfo?.tracker ?? UNKNOWN_TRACKER;
+		if (tracker === UNKNOWN_TRACKER) {
+			summary.FOUND_BAD_FORMAT = true;
+		}
+
+		const { fullMatches, partialMatches, foundBlocked } =
+			await injectDecideStage(meta, searchees);
+		if (fullMatches.length === 0 && partialMatches.length === 0) {
+			if (foundBlocked) {
+				logger.info({
+					label: Label.INJECT,
+					message: `${progress} ${metaLog} ${chalk.yellow("possibly blocklisted")}: ${filePathLog}`,
+				});
+				summary.BLOCKED++;
+			} else {
+				logger.info({
+					label: Label.INJECT,
+					message: `${progress} ${metaLog} ${chalk.red("has no matches")}: ${filePathLog}`,
+				});
+				summary.UNMATCHED++;
+			}
+			continue;
+		}
+
+		const {
+			injectionResult,
+			matchedSearchee,
+			matchedDecision,
+			linkedNewFiles,
+		} = await injectInitialAction(
+			meta,
+			fullMatches,
+			partialMatches,
+			tracker,
+		);
+		if (injectionResult === InjectionResult.FAILURE) {
+			logger.error({
+				label: Label.INJECT,
+				message: `${progress} Failed to inject ${filePathLog} - ${chalk.red(injectionResult)}`,
+			});
+			summary.FAILED++;
+			toDelete.delete(torrentFilePath);
+			continue;
+		}
+		if (injectionResult === InjectionResult.TORRENT_NOT_COMPLETE) {
+			if (
+				!linkDir ||
+				fs.statSync(torrentFilePath).mtimeMs >= Date.now() - ms("1 day")
+			) {
+				// Normal case where source is likely still downloading
+				logger.warn({
+					label: Label.INJECT,
+					message: `${progress} Unable to inject ${filePathLog} - ${chalk.yellow(injectionResult)}`,
+				});
+			} else {
+				// Since source is stalled, add to client paused so user can resume later if desired
+				// Try linking all possible matches as they may have different files
+				await injectFromStalledTorrent(
+					meta,
+					fullMatches,
+					partialMatches,
+					tracker,
+					injectionResult,
+					progress,
+					filePathLog,
+				);
+			}
+			summary.INCOMPLETE_SEARCHEES++;
+			continue;
+		}
+		const result = await getClient().isTorrentComplete(meta.infoHash);
+		let isComplete = result.isOk() ? result.unwrap() : false;
+		if (isComplete) {
+			toDelete.add(torrentFilePath);
```

Comment:
I'll implement this if it's not `outputDir` but otherwise we should keep it. Torrents saved long ago or have it's sources removed before the job would stay around forever unless the user manually intervenes. After this is merged, a user should never interact with `outputDir`.

Reply:
> Torrents saved long ago or have its sources removed before the job would stay around forever unless the user manually intervenes. 

why is it important to deal with those? We also keep torrents in the torrent_cache. 

I don't think deleting old torrents is in the critical path of this feature delivering value to users.

Reply:
I don't see why we shouldn't be deleting them. If they are in `outputDir` for more than a week, then it means the searchee has been deleted and will never be injected. There is no valid use for these torrents. Every time the job is ran, there will be hundreds or thousands of attempts constantly only getting worse with time.

The docs for this feature makes it clear that users should no longer be interacting with `outputDir`, I don't see any reason to keep these torrents.

Reply:
I tend to agree on this, and assuming that if action is not set to `save` there is no reason why a user would typically need those files past any particular date after a week of failing to inject.

Am i missing something?

Reply:
> there will be hundreds or thousands of attempts constantly only getting worse with time.

These attempts are cheap. They don't do any tracker IO, and "hundreds or thousands" doesn't really pass the sniff test to me. I am the oldest user of cross-seed and my outputDir has 17 torrents in it. (they look like they're all from v6 as well, did the old ones get rewritten to the new format or is v6 just less reliable?)

There's no guarantee that `cross-seed` owns every torrent in `outputDir`, and it **cannot** make destructive assumptions about torrents it doesn't own. We experienced this with Sonarr and qBittorrent - Sonarr assumes that it owns every torrent with that category, even though we _have_ to use categories to store the files in the right place. That forced us to implement a hacky workaround. 

Here's what I'll propose—we delete files if they look like cross-seed created them. That is, they follow the `[mediaType][tracker]torrentname[infoHash].torrent` format. I pushed it to a separate branch because I'm not sure if I got all the logic right. https://github.com/cross-seed/cross-seed/pull/770/files

Reply:
The only scenarios deletion will happen now is: if no matches are found, if it's blocklisted, or if it's in client and complete.

In the case of a stalled injection or source, the user will now need to remove the stalled torrent before it gets deleted.

Injection failures won't cause deletion as it should be temporary or user error.

Reply:
> if no matches are found, if it's blocklisted, or if it's in client and complete.

Can we just do the third one in this PR?

Reply:
Users would need to manually clean up outputDir after deleting stalled torrents or if they use a blocklist. Remember that these torrents only exist in outputDir because injection failed. In an ideal world, these torrents wouldn't have been saved in the first place. So it feels like creating more work for the user for no real gain.